addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/GuessParamsThread.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.paramminer;
+
+import java.util.Date;
+import javax.swing.ListModel;
+import org.apache.commons.lang.time.StopWatch;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.model.ScanListenner;
+import org.zaproxy.zap.model.ScanThread;
+import org.zaproxy.zap.model.TechSet;
+import org.zaproxy.zap.users.User;
+
+public class GuessParamsThread extends ScanThread implements GuesserListner {
+
+    private boolean stopScan = false;
+    private boolean isPaused = false;
+    private boolean isAlive = false;
+
+    private ExtensionParamMiner extension;
+    private int guessDone = 0;
+    private int guessTodo = 0;
+
+    private static final Logger log = LogManager.getLogger(GuessParamsThread.class);
+
+    private HttpMessage message;
+    private final String id;
+    private StopWatch stopWatch;
+    private User scanUser;
+
+    private ParamGuesser paramGuesser;
+
+    public GuessParamsThread(
+            ExtensionParamMiner extension,
+            HttpMessage message,
+            String id,
+            String site,
+            ScanListenner scanListenner) {
+        super(site, scanListenner);
+        this.id = id;
+        this.extension = extension;
+        this.message = message;
+        this.site = site;
+        log.debug("GuessParamsThread created for {}", site);
+    }
+
+    public long getTimeTakenInMs() {
+        return stopWatch.getTime() - stopWatch.getStartTime();
```

Comment:
```
ong [org](eclipse-javadoc:%E2%98%82=zap/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/org.apache.commons%5C/commons-lang3%5C/3.12.0%5C/c6842c86792ff03b9f1d1fe2aab8dc23aa6c6f0e%5C/commons-lang3-3.12.0.jar=/gradle_used_by_scope=/main,test,testGui=/%3Corg).[apache](eclipse-javadoc:%E2%98%82=zap/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/org.apache.commons%5C/commons-lang3%5C/3.12.0%5C/c6842c86792ff03b9f1d1fe2aab8dc23aa6c6f0e%5C/commons-lang3-3.12.0.jar=/gradle_used_by_scope=/main,test,testGui=/%3Corg.apache).[commons](eclipse-javadoc:%E2%98%82=zap/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/org.apache.commons%5C/commons-lang3%5C/3.12.0%5C/c6842c86792ff03b9f1d1fe2aab8dc23aa6c6f0e%5C/commons-lang3-3.12.0.jar=/gradle_used_by_scope=/main,test,testGui=/%3Corg.apache.commons).[lang3](eclipse-javadoc:%E2%98%82=zap/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/org.apache.commons%5C/commons-lang3%5C/3.12.0%5C/c6842c86792ff03b9f1d1fe2aab8dc23aa6c6f0e%5C/commons-lang3-3.12.0.jar=/gradle_used_by_scope=/main,test,testGui=/%3Corg.apache.commons.lang3).[time](eclipse-javadoc:%E2%98%82=zap/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/org.apache.commons%5C/commons-lang3%5C/3.12.0%5C/c6842c86792ff03b9f1d1fe2aab8dc23aa6c6f0e%5C/commons-lang3-3.12.0.jar=/gradle_used_by_scope=/main,test,testGui=/%3Corg.apache.commons.lang3.time).[StopWatch](eclipse-javadoc:%E2%98%82=zap/C:%5C/Users%5C/thorin%5C/.gradle%5C/caches%5C/modules-2%5C/files-2.1%5C/org.apache.commons%5C/commons-lang3%5C/3.12.0%5C/c6842c86792ff03b9f1d1fe2aab8dc23aa6c6f0e%5C/commons-lang3-3.12.0.jar=/gradle_used_by_scope=/main,test,testGui=/%3Corg.apache.commons.lang3.time(StopWatch.class%E2%98%83StopWatch).getTime()



Gets the time on the stopwatch. 

This is either the time between the start and the moment this method is called, or the amount of time between start and stop. 

Returns: the time in milliseconds
```

Reply:
Do I have to use lang3 tho?

Reply:
Good question. I'm not sure. But lang3 is included in core so it shouldn't be a big deal.

Reply:
I used lang for the StopWatch ... build looks happy so I'm happy with it :)

Reply:
My point was more that you didn't need to do the math here. getTime() returns elapsed.

Reply:
Yep did that :). Will push the changes later. With some more additions :)